Cron/UpdateProductInstallmentValues.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace RicardoMartins\PagSeguro\Cron;
+
+use Magento\Framework\Phrase;
+
+class UpdateProductInstallmentValues
+{
+    protected $_productFactory;
+    protected $_stockFilter;
+    protected $pagSeguroHelper;
+    protected $_scopeConfig;
+
+
+    public function __construct(
+        \Magento\Catalog\Model\ResourceModel\Product\CollectionFactory $productFactory,
+        \Magento\CatalogInventory\Helper\Stock $stockFilter,
+        \RicardoMartins\PagSeguro\Helper\Data $pagSeguroHelper,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+    )
+    {
+        $this->_productFactory = $productFactory;
+        $this->_stockFilter = $stockFilter;
+        $this->pagSeguroHelper = $pagSeguroHelper;
+        $this->_scopeConfig = $scopeConfig;
+    }
+    public function execute()
+    {
+//        echo "Iniciando...\n";
+        if(!$this->_scopeConfig->getValue('payment/rm_pagseguro_cc/show_installments_product_page',
+            \Magento\Store\Model\ScopeInterface::SCOPE_WEBSITE)) return;
+
+            $collection = $this->_productFactory->create();
+        $collection->addAttributeToSelect('*');
+        $collection->addAttributeToFilter('rm_pagseguro_last_update',0);
+        $collection->addAttributeToFilter('status',\Magento\Catalog\Model\Product\Attribute\Source\Status::STATUS_ENABLED);
+        $collection->setPageSize(15)->setCurPage(1)->load();
+        if(!count($collection)) return;
+        $params['sessionId'] = $this->pagSeguroHelper->getSessionId();
+        foreach($collection as $product){
+//            echo "Produto: ".$product->getName()."\n";
+            $price = $product->getFinalPrice();
+//            echo "Preco: ".$product->getFinalPrice()."\n";
+//            echo "rm_pagseguro_last_update: '".$product->getRmPagseguroLastUpdate()."'\n";
+            $params['amount'] = number_format($price,2,".","");
+
+            if($this->pagSeguroHelper->isSandbox()) {
+                $url = "https://ws.sandbox.pagseguro.uol.com.br/v2/installments.json";
+            }
+            else{
+                $url = "https://ws.pagseguro.uol.com.br/checkout/v2/installments";
```

Comment:
This URL is invalid and will return a temporary redirect (307) which will return an error.
Just add a .json in the end and it should be fine.